huaweicloud/services/waf/resource_huaweicloud_waf_rule_anti_crawler.go

```diff
@@ -0,0 +1,349 @@
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceRuleAntiCrawler() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRuleCreate,
+		UpdateContext: resourceRuleUpdate,
+		ReadContext:   resourceRuleRead,
+		DeleteContext: resourceRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceWAFRuleImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the policy ID.`,
+			},
+			"protection_mode": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the protection mode of WAF anti crawler rule.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the rule name.`,
+			},
+			"priority": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: `Specifies the priority.`,
+			},
+			"conditions": {
+				Type:        schema.TypeList,
+				Elem:        ruleAntiCrawlerConditionsSchema(),
+				Required:    true,
+				Description: `Specifies the match condition list.`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the enterprise project ID.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the rule description.`,
+			},
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The rule status.`,
+			},
+		},
+	}
+}
+
+func ruleAntiCrawlerConditionsSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"field": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the field type.`,
+			},
+			"logic": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the logic for matching the condition.`,
+			},
+			"content": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the content of the condition.`,
+			},
+			"reference_table_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the reference table ID.`,
+			},
+		},
+	}
+}
+
+func updateAntiCrawlerProtectionMode(client *golangsdk.ServiceClient, d *schema.ResourceData,
+	cfg *config.Config) error {
+	updatePath := client.Endpoint + "v1/{project_id}/waf/policy/{policy_id}/anticrawler"
```

Comment:
those lines are the same as lines in resourceRuleCreate

Reply:
OK, I have removed the duplicate codes.